pkg/snapshot/snapshotter/snapshotter_test.go

```diff
@@ -570,192 +543,299 @@ var _ = Describe("Snapshotter", func() {
 
 				validateLimitBasedSnapshots(list, maxBackups, snapsInV1)
 			})
-		})
-	})
 
-	Describe("Scenarios to take full-snapshot during startup of backup-restore", func() {
-		var (
-			ssr                    *Snapshotter
-			currentMin             int
-			currentHour            int
-			fullSnapshotTimeWindow float64
-		)
-		BeforeEach(func() {
-			fullSnapshotTimeWindow = 24
-			currentHour = time.Now().Hour()
-			currentMin = time.Now().Minute()
-			snapstoreConfig = &brtypes.SnapstoreConfig{Container: path.Join(outputDir, "default.bkp")}
-			store, err = snapstore.GetSnapstore(snapstoreConfig)
-			Expect(err).ShouldNot(HaveOccurred())
-		})
-		Context("No previous snapshot was taken", func() {
-			It("should return true", func() {
-				snapshotterConfig := &brtypes.SnapshotterConfig{
-					FullSnapshotSchedule: fmt.Sprintf("%d %d * * *", (currentMin+1)%60, (currentHour+2)%24),
-				}
+			Describe("###GarbageCollectDeltaSnapshots", func() {
+				const (
+					deltaSnapshotCount = 6
+					testDir            = "garbagecollector_deltasnapshots.bkp"
+				)
 
-				ssr, err = NewSnapshotter(logger, snapshotterConfig, store, etcdConnectionConfig, compressionConfig, healthConfig, snapstoreConfig)
-				Expect(err).ShouldNot(HaveOccurred())
+				var snapshotterConfig *brtypes.SnapshotterConfig
 
-				// No previous snapshot was taken
-				ssr.PrevFullSnapshot = nil
-				isFullSnapMissed := ssr.IsFullSnapshotRequiredAtStartup(fullSnapshotTimeWindow)
-				Expect(isFullSnapMissed).Should(BeTrue())
-			})
-		})
-		Context("If previous snapshot was final full snapshot", func() {
-			It("should return true", func() {
-				snapshotterConfig := &brtypes.SnapshotterConfig{
-					FullSnapshotSchedule: fmt.Sprintf("%d %d * * *", (currentMin+1)%60, (currentHour+2)%24),
-				}
+				BeforeEach(func() {
+					snapshotterConfig = &brtypes.SnapshotterConfig{
+						FullSnapshotSchedule:     schedule,
+						DeltaSnapshotPeriod:      wrappers.Duration{Duration: 10 * time.Second},
```

Comment:
Technically, the delta snapshot period according to the `populateDeltaSnapshot()` function would be `10 minutes`.

Reply:
Fixed as part of https://github.com/gardener/etcd-backup-restore/pull/640/commits/182d3dd0546a1649902a65d9d52f34fabc9e0e64